adapter-excel/src/main/java/ch/prevo/open/node/adapter/excel/ErrorWriter.java

```diff
@@ -0,0 +1,181 @@
+
+import javax.validation.ConstraintViolation;
+import java.io.Closeable;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.ADDITIONAL_NAME_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.CITY_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.DATE_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.IBAN_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.NAME_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.OASI_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.POSTAL_CODE_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.REFERENCE_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.RETIREMENT_FUND_UID_COLUMN_INDEX;
+import static ch.prevo.open.node.adapter.excel.ExcelConstants.STREET_COLUMN_INDEX;
+
+public class ErrorWriter implements Closeable {
+
+    private final Workbook workbook;
+    private final String filename;
+    private final CreationHelper factory;
+
+    private final CellStyle errorStyle;
+    private final CellStyle errorDateStyle;
+
+    public ErrorWriter(Workbook workbook, String filename) {
+        this.workbook = workbook;
+        this.filename = filename;
+        this.factory = workbook.getCreationHelper();
+
+        this.errorStyle = workbook.createCellStyle();
+        errorStyle.setFillForegroundColor(IndexedColors.YELLOW.getIndex());
+        errorStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+
+        this.errorDateStyle = workbook.createCellStyle();
+        final short shortDateFormat = workbook.getCreationHelper().createDataFormat().getFormat("d-mmm-yy");
+        errorDateStyle.setDataFormat(shortDateFormat);
+        errorDateStyle.setFillForegroundColor(IndexedColors.YELLOW.getIndex());
+        errorDateStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+    }
+
+    public void markTerminationErrors(Map<EmploymentTermination, Set<ConstraintViolation<EmploymentTermination>>> violationSet) {
+        final Sheet sheet = workbook.getSheet(ExcelConstants.TERMINATION_LABEL);
+        final Drawing<?> drawing = sheet.createDrawingPatriarch();
+        violationSet.forEach(
+                (termination, violations) -> {
+                    final Row row = findRow(sheet, termination)
+                            .orElseThrow(() -> new IllegalStateException("Violation received which does not have a matching entry in the input-table " + termination));
+                    for (final ConstraintViolation<EmploymentTermination> violation : violations) {
+                        final String propertyPath = violation.getPropertyPath().toString();
```

Comment:
Should we make it case-insensitive? Since refactor-rename most likely wouldn't find this.

Reply:
I don't think that making it case-insensitive would add much value. But I would be happy to hear any suggestion to stabilize this against refactorings.

Reply:
Fair enough - since a renaming wouldn't be caught... well having a test would help of course. Currently I don't have a much better idea